File: content/api-documentation/how-to/orders/examples/placeTrailingStopOrder.go
package main

import (
	"github.com/financefeasthq/financefeast-trade-api-go/financefeast"
	"github.com/shopspring/decimal"
)

func init() {
	financefeast.SetBaseUrl("https://paper-api.financefeast.markets")
}

func main() {
	// Submit a market order to buy 1 share of Apple at market price
	symbol := "AAPL"
	financefeast.PlaceOrder(financefeast.PlaceOrderRequest{
		AssetKey: &symbol,
		Qty: decimal.NewFromFloat(1),
		Side: financefeast.Buy,
		Type: financefeast.Market,
		TimeInForce: financefeast.Day,
	})

	// Submit a trailing stop order to sell 1 share of Apple at a
    // trailing stop of
	symbol = "AAPL"
	financefeast.PlaceOrder(financefeast.PlaceOrderRequest{
		AssetKey: &symbol,
		Qty: decimal.NewFromFloat(1),
		Side: financefeast.Sell,
		Type: financefeast.TrailingStop,
		StopPrice: decimal.NewFromFloat(1.00),  // stop price will be hwm - 1.00$
		TimeInForce: financefeast.Day,
	})

	// Alternatively, you could use trail_percent:
	symbol = "AAPL"
	financefeast.PlaceOrder(financefeast.PlaceOrderRequest{
		AssetKey: &symbol,
		Qty: decimal.NewFromFloat(1),
		Side: financefeast.Sell,
		Type: financefeast.TrailingStop,
		TrailPercent: decimal.NewFromFloat(1.0),  // stop price will be hwm*0.99
		TimeInForce: financefeast.Day,
	})
}

File: content/api-documentation/how-to/orders/examples/placeTrailingStopOrder.js
const financefeast = require('@financefeasthq/financefeast-trade-api')
const financefeast = new financefeast()

// Submit a market order to buy 1 share of Apple at market price
financefeast.createOrder({
    symbol: 'AAPL',
    qty: 1,
    side: 'buy',
    type: 'market',
    time_in_force: 'day'
})

// Submit a trailing stop order to sell 1 share of Apple at a
// trailing stop of
financefeast.createOrder({
    symbol: 'AAPL',
    qty: 1,
    side: 'sell',
    type: 'trailing_stop',
    trail_price: 1.00,  // stop price will be hwm - 1.00$
    time_in_force: 'day',
})

// Alternatively, you could use trail_percent:
financefeast.createOrder({
    symbol: 'AAPL',
    qty: 1,
    side: 'sell',
    type: 'trailing_stop',
    trail_percent: 1.0,  // stop price will be hwm*0.99
    time_in_force: 'day',
})